lib/sdotypes.c

```diff
@@ -993,6 +993,27 @@ sdo_string_t *sdo_string_alloc(void)
 	return (sdo_string_t *)sdo_alloc(sizeof(sdo_string_t));
 }
 
+/**
+ * Create sdo_string_t object with the given size.
+ * @return an allocated sdo_string_t object
+ */
+sdo_string_t *sdo_string_alloc_size(size_t byte_sz) {
```

Comment:
What is the expectation if byte_sz = 0? Currently it returns a valid (sdo_string_t*)s, but the s->bytes=NULL.

Reply:
Updated